### DISCUSSION TITLE
exporting otel metrics via grpc (tonic)  using a proxy

### DISCUSSION THREAD

**Discussion by aheusser** [open-telemetry/opentelemetry-rust#1589] [Q&A]
I have been able to successfully export OTEL metrics **through an enterprise proxy** to an Http  OTEL endpoint by creating a `MeterProvider` using the `.with_http_client()` member of the `HttpExporterBuilder` ... where the client has been preconfigured with the proxy to use.

Now I am trying to accomplish a similar goal to a Grpc endpoint. I have **basic** Grpc exporting working by creating a `MeterProvider` using the `TonicExporterBuilder` ... but I cannot find an API-supported way, or any example code, to configure the tonic Grpc client to use and enterprise proxy to communicate with the Grpc OTEL endpoint.

Is this a problem that has already been tackled? Does anyone have an example they can share of how to do it using `opentelemetry-rust`?

I really want to use Grpc for payload efficiency, but if it is not feasible to optionally configure it to go through an enterprise proxy, then it is going to be of limited utility for me.

Thanks, in advance, for any insights!

**Comment by cxw620** [comment]
Tonic does not support any proxy offcially and we have to implement a proxy connector manually, see [my comment here](https://github.com/hyperium/tonic/discussions/1206) as an example and try fork and modify codes of `opentelemetry-rust`, or submit an issue for help.
